tests/unit/s2n_send_test.c

```diff
@@ -463,5 +463,88 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(conn->out.blob.size, out_size[mfl]);
     }
 
+    /* Test dynamic record threshold */
```

Comment:
Would it make sense to set `timeout_threshold` to a small value, sleep for a bit after the fourth send, and send again to make sure the record size reduces to a single MTU again?

I'm not sure if sleeping in a test is weird or not. If it is, we could maybe manually alter 
`conn->dynamic_record_timeout_threshold` or `conn->last_write_elapsed` after the fourth send to simulate time passing. Or is this aspect not that important to test anyway?

Reply:
I'm more interested in the record sizes than the timeout behavior, but I should make that clearer in the test name.